templates/simple/simple.sh

```diff
@@ -167,20 +167,9 @@ cat << '_HEREDOC_' > ${JSB_TMP}/ifdh.sh
 which ifdh > /dev/null 2>&1
 has_ifdh=$?
 if [ "$has_ifdh" -ne "0" ] ; then
-    unset PRODUCTS
-    for setup_file in /cvmfs/fermilab.opensciencegrid.org/products/common/etc/setups /grid/fermiapp/products/common/etc/setups.sh /fnal/ups/etc/setups.sh ; do
-      if [ -e "$setup_file" ] && [ "$has_ifdh" -ne "0" ]; then
-         source $setup_file
-         ups exist ifdhc $IFDH_VERSION
-         has_ifdh=$?
-         if [ "$has_ifdh" = "0" ] ; then
-             setup ifdhc $IFDH_VERSION
-             break
-         else
-            unset PRODUCTS
-         fi
-     fi
-   done
+    . /cvmfs/fermilab.opensciencegrid.org/packages/common/setup-env.sh
+    sos=$(spack arch --operating-system)
+    spack env activate jobsub_env_${sos}_${IFDH_VERSION:-current}
```

Comment:
A few questions:

1. Is this environment already created?  
2. Do we need to bring whatever configuration is behind that environment into the jobsub_lite repo somehow?
3. Am I reading line 172 correct that if $IFDH_VERSION is not set, then the environment to be activated will be something like `jobsub_env_scientific7_current`?  Do we have an AL9 environment ready to go too?

Reply:
dunegpvm02>$ . /cvmfs/fermilab.opensciencegrid.org/packages/common/setup-env.sh

<dunegpvm02>$ spack env list
==> 6 environments
    jobsub_env_almalinux9_current  jobsub_env_scientific7_current
    jobsub_env_almalinux9_v2_6_20  jobsub_env_scientific7_v2_6_20
    jobsub_env_almalinux9_v2_7     jobsub_env_scientific7_v2_7

The environments in question just have ifdhc and dependencies (mainly python).
Note that the `*_current` environments are just symlinks to the particular versions.

And as usual, these are active only in the subshell of the ifdh.sh script, so as not to mess with any experiment job environments.

Reply:
Awesome - thanks for clearing that up for me.  So you think that we don't need to version-control that here?

I'll wait until tomorrow to merge this, since I'd rather not do it while we're deploying 1.6 in case of a hypotethical emergency bugfix we'd need to do.

Reply:
Yes.
As to environments,  I can add new ones as part of releasing ifdhc, and we can move the _current links when suitably announced etc. as we do now when making ifhdc current in UPS.   I mean, I could add a script to the jobsub repository that creates the environments for a given version?   It's just:
<pre>
ename=jobsub_env_${sos}_${version}
spack env create $ename
spack -e $ename add ifdhc/version-hash
spack -e $ename install
</pre>
for each new spack instance...

Reply:
Marc and I discussed this offline - mostly about responsibilities (jobsub vs IFDH).  We decided on the following:

1.  The spack environment name will be `ifdh_env_${sos}_${version}` or `ifdh_env_${sos}_current`.  This is so that (a) The name implies that the spack environment is maintained by the IFDHC project, not by the jobsub project; and (b) Marc anticipates that users can be instructed to use that spack environment to select versions of IFDHC, while testing, for example.
2. When the IFDHC developers create a new release of IFDHC, it will be their responsibility to create a new spack environment and to change the `current` environment to point to the new version, on their timeline.
3. This way, when a new IFDHC has been released _for testing_, jobsub will continue to use the old spack environment (still marked `current`) by default.  When the IFDHC developers are satisfied with the test release, then they will change the `current` spack environment to use that new release.

As far as this PR is concerned, two things need to happen:
1. Change line 172 to be `spack env activate ifdh_env_${sos}_${IFDH_VERSION:-current}`
2. Change the current spack `jobsub_env*` environments to be `ifdh_env_*`

After those two are done, we can merge this PR.